int16 sx
int16 sy
int16 gx
int16 gy
